Description: set x100 default volume
 降低Output频道默认音量，
 厂商反馈音量太高，会导致音频信号线电压太高，
 会降低喇叭的寿命，出现烧喇叭的可能。
 所以这里遵循厂商的建议，将音量设置为-2dB。
Author: liduanjun <liduanjun@uniontech.com>

Origin: https://gerrit.uniontech.com/plugins/gitiles/base/alsa-utils/+/6dcf066a91aa4a1384337cfd48662b1bc7a33b3d
Bug: https://pms.uniontech.com/zentao/bug-view-98898.html
Last-Update: 2022-05-19

--- alsa-utils-1.2.6.orig/alsactl/init/default
+++ alsa-utils-1.2.6/alsactl/init/default
@@ -36,6 +36,12 @@ CTL{name}="Right Mixer Playback Switch",
   CTL{values}="on"
 
 CTL{reset}="mixer"
+CTL{name}="Output 1 Playback Volume",CTL{do_search}=="1", \
+  CTL{write}!="-2dB",CTL{values}="$env{ppercent}"
+CTL{name}="Output 2 Playback Volume",CTL{do_search}=="1", \
+  CTL{write}!="-2dB",CTL{values}="$env{ppercent}"
+
+CTL{reset}="mixer"
 CTL{name}="Master Front Playback Volume",CTL{do_search}=="1", \
   ENV{has_pmaster_vol}:="true", \
   CTL{write}!="$env{pvolume}",CTL{values}="$env{ppercent}"
