Author: Elimar Riesebieter <riesebie@lxtec.de>
Description: fix some spell errors in alsactl binary.

Index: alsa-utils/alsactl/alsactl.c
===================================================================
--- alsa-utils.orig/alsactl/alsactl.c
+++ alsa-utils/alsactl/alsactl.c
@@ -96,7 +96,7 @@ static struct arg args[] = {
 { FILEARG | 'e', "pid-file", "pathname for the process id (daemon mode)" },
 { HEADER, NULL, "Available init options:" },
 { ENVARG | 'E', "env", "set environment variable for init phase (NAME=VALUE)" },
-{ FILEARG | 'i', "initfile", "main configuation file for init phase" },
+{ FILEARG | 'i', "initfile", "main configuration file for init phase" },
 { 0, NULL, "  (default " DATADIR "/init/00main)" },
 { 'b', "background", "run daemon in background" },
 { 's', "syslog", "use syslog for messages" },
Index: alsa-utils/alsactl/init_parse.c
===================================================================
--- alsa-utils.orig/alsactl/init_parse.c
+++ alsa-utils/alsactl/init_parse.c
@@ -1209,7 +1209,7 @@ found:
 		/* possibly truncate to format-char specified length */
 		if (len != -1) {
 			head[len] = '\0';
-			dbg("truncate to %i chars, subtitution string becomes '%s'", len, head);
+			dbg("truncate to %i chars, substitution string becomes '%s'", len, head);
 		}
 		strlcat(string, temp, maxsize);
 	}
Index: alsa-utils/alsactl/state.c
===================================================================
--- alsa-utils.orig/alsactl/state.c
+++ alsa-utils/alsactl/state.c
@@ -1168,7 +1168,7 @@ static int restore_config_value(snd_ctl_
 	case SND_CTL_ELEM_TYPE_IEC958:
 		break;
 	default:
-		cerror(doit, "Unknow control type: %d", type);
+		cerror(doit, "Unknown control type: %d", type);
 		return -EINVAL;
 	}
 	return 0;
